docs/federation.md

```diff
@@ -0,0 +1,256 @@
+```java
+@DgsEntityFetcher(name = "Show")
+public Show show(Map<String, Object> values) {
+        return new Show((String) values.get("id"), null);
+}
+```
+
+!!!tip
+    Remember that the Show Java type here is generated by codegen.
+    It's generated from the schema, so it only has the fields our schema specifies.
+
+### Providing Data with a Data Fetcher
+
+Now the DGS knows how to create a Show instance when an `_entities` query is received, we can specify how to hydrate data for the reviews field.
+
+[Full code](https://github.com/Netflix/dgs-federation-example/blob/master/reviews-dgs/src/main/java/com/example/demo/datafetchers/ReviewsDatafetcher.java#L37)
+```java
+@DgsData(parentType = "Show", field = "reviews")
+public List<Review> reviews(DgsDataFetchingEnvironment dataFetchingEnvironment)  {
+    Show show = dataFetchingEnvironment.getSource();
+    return reviews.get(show.getId());
+}
+```
+
+### Testing a Federated Query
+
+You can always manually test federated queries by running the gateway and your DGS locally. 
+You can also manually test a federated query against just your DGS, without the gateway, using the `_entities` query to replicate the call made to your DGS by the gateway.
+
+For automated tests, the [QueryExecutor](./query-execution-testing.md) gives a way to run queries from unit tests, with very little startup overhead (in the order of 500ms).
+We can capture (or manually write) the `_entities` query that the gateway sends to the DGS.
+When running the query through the (locally running) gateway, the DGS will log the query that it receives.
+Simply copy this query in a `QueryExecutor` test, and that verifies the DGS in isolation.
+
+```java
+@SpringBootTest(classes = {DgsAutoConfiguration.class, ReviewsDatafetcher.class})
+class ReviewssDatafetcherTest {
+
+    @Autowired
+    DgsQueryExecutor dgsQueryExecutor;
+
+    @Test
+    void shows() {
+        Map<String,Object> representation = new HashMap<>();
+        representation.put("__typename", "Show");
+        representation.put("id", "1");
+        List<Map<String, Object>> representationsList = new ArrayList<>();
+        representationsList.add(representation);
+
+        Map<String, Object> variables = new HashMap<>();
+        variables.put("representations", representationsList);
+        List<Review> reviewsList = dgsQueryExecutor.executeAndExtractJsonPathAsObject(
+                "query ($representations:[_Any!]!) {" +
+                        "_entities(representations:$representations) {" +
+                        "... on Show {" +
+                        "   reviews {" +
+                        "       starRating" +
+                        "}}}}",
+                "data['_entities'][0].reviews", variables, new TypeRef<>() {});
+
+        assertThat(reviewsList).isNotNull();
+        assertThat(reviewsList.size()).isEqualTo(3);
```

Comment:
nit, you can do `assertThat(reviewsList).hasSize(3)` or even chain with 
```
  assertThat(reviewsList)
       .isNotNull()
       .hasSize(3)
```

Reply:
Done